crates/cranelift/src/func_environ.rs

```diff
@@ -212,6 +224,73 @@ impl<'module_environment> FuncEnvironment<'module_environment> {
     fn vmctx(&mut self, func: &mut Function) -> ir::GlobalValue {
         self.vmctx.unwrap_or_else(|| {
             let vmctx = func.create_global_value(ir::GlobalValueData::VMContext);
+            if self.isa.flags().enable_pcc() {
+                let mut fields = vec![];
+                for (memory_idx, plan) in &self.module.memory_plans {
+                    // For now, we only do PCC on defined, owned (non-shared), static, 32-bit memories (whew!).
+                    if plan.memory.shared || plan.memory.memory64 {
+                        continue;
+                    }
+                    let static_bound_wasm_pages = match &plan.style {
+                        MemoryStyle::Dynamic { .. } => continue,
+                        MemoryStyle::Static { bound } => *bound,
+                    };
+                    let static_bound = static_bound_wasm_pages.checked_mul(0x1_0000).unwrap();
+                    let def_memory_idx = match self.module.defined_memory_index(memory_idx) {
+                        Some(i) => i,
+                        None => continue,
+                    };
+                    let owned_memory_idx = self.module.owned_memory_index(def_memory_idx);
+
+                    let base_field_offset = self
+                        .offsets
+                        .vmctx_vmmemory_definition_base(owned_memory_idx);
+                    // Create a new "blob" memory type to represent the pointed-to memory.
+                    let data_mt = func.create_memory_type(ir::MemoryTypeData::Memory {
+                        size: static_bound
+                            .checked_add(plan.offset_guard_size)
+                            .expect("Memory plan has overflowing size plus guard"),
+                    });
+                    log::trace!(
+                        "data_mt: static bound is {:x}, offset guard size is {:?}",
+                        static_bound,
+                        plan.offset_guard_size
+                    );
```

Comment:
Similarly, I don't think this trace will be useful long term.

Reply:
Removed!